src/context.rs

```diff
@@ -182,8 +182,19 @@ impl Game {
                 self.joypad.borrow_mut().deref_mut(),
                 self.dma.borrow_mut().deref_mut()
             );
-            self.cycle_count += 1;
             self.check_scheduled_stop(!frame_not_finished);
+            #[cfg(feature = "cgb")]
+            if self.cpu.io_regs.borrow().fast_mode() {
+                cycles!(
+                    self.clock,
+                    &mut self.addr_bus,
+                    &mut self.cpu,
+                    self.timer.borrow_mut().deref_mut(),
+                    self.dma.borrow_mut().deref_mut()
+                );
+                self.check_scheduled_stop(!frame_not_finished);
+            }
+            self.cycle_count += 1;
```

Comment:
missing one increment of `cycle_count` in fast mode

Reply:
I don't think because in double speed the cpu & co (excluding the ppu) are ticked 2 time in 1 cycle

Reply:
Ok makes sense. The nomenclature might be a little confusing then between the main cycle and the sub-cycles but it's ok.